echopype/utils/coding.py

```diff
@@ -56,7 +56,7 @@ def set_encodings(ds: xr.Dataset) -> xr.Dataset:
     for var, encoding in DEFAULT_ENCODINGS.items():
         if var in new_ds:
             da = new_ds[var].copy()
-            if "_time" in var:
+            if "_time" in var or ("time" in var and len(var) == 5):
```

Comment:
what is this `len(var) == 5` thing for here?

Reply:
The second test condition (in parenthesis) is testing for names like `time1`, `time2` ... `time9`. Actually, I'm wondering if we can just change the whole if-statement to `if "time" in var:`. I don't remember if we were testing for "_time" rather than "time" for a specific reason. I was being cautious in this PR.

Reply:
> Agreed about the intention being more explicit.

It is also that testing a length of 5 would now allow more than one digit.

> It would work. It's testing for the presence of the "time" string in var, not an exact match. 

Oh right it is "in" rather than "is" in the if statement. 

I think the intention was trying to test everywhere time encodings occur. With that in mind using `time` would work. But, I think for testing it'd be better to be more specific by allowing `*_time` and `time*` patterns.

Reply:
> I think the intention was trying to test everywhere time encodings occur. With that in mind using time would work. But, I think for testing it'd be better to be more specific by allowing *_time and time* patterns.

I see two options. 

Given that we intend to capture all variables that include the "time" substring (right?), my preference is the simple `in` test plus a comment. The code is super clean, and the comment conveys the intent.

```python
# Match all variable names containing a "time" substring, including *_time and time*
if "time" in var:
```

Here's one option using a regex. The complexity of the regex depends on how strict you want to be. While the regex makes the intent precisely coded, I think it makes it less immediately readable. Plus using regex adds a slight performance hit (probably trivial, though):

```python
from re import search

# Match all variable names matching the patterns *_time or time*
if bool(search(r".+_time$|^time[\d]+", var)):

# an alternative that might be more readable:
if bool(search(r".+_time$", var)) or if bool(search(r"^time[\d]+", var)):
```

I think the crux is whether we deliberately want to limit the test to `*_time` and `time*`

Reply:
Thanks @emiliom for putting these together. My vote is to limit the tests to `*_time` and `time*`, to avoid accidental error, if there happen to be some variables that has "time" in it but is actually not a _time_ per se.

Also, I just realized that somehow I thought this was in tests. These are not, so all the more important to be accidental-error proof!

Reply:
I've pushed a solution. FYI, AD2CP uses some time variable that must be of the form `*_time*`, so my final regex looks like this:
```python
bool(search(r"_time|^time[\d]+$", var))
```
The first part is basically the same as what we had previously (equivalent to `"_time" in var`)